src/main/java/com/example/springcrud/service/CommentService.java

```diff
@@ -0,0 +1,40 @@
+package com.example.springcrud.service;
+
+import com.example.springcrud.dto.CommentDto;
+import com.example.springcrud.entity.CommentEntity;
+import com.example.springcrud.entity.ProductEntity;
+import com.example.springcrud.repository.CommentRepository;
+import com.example.springcrud.repository.ProductRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+public class CommentService {
+
+    private final CommentRepository commentRepository;
+    private final ProductRepository productRepository;
+
+    public List<CommentEntity> getComments() {
+        return commentRepository.findAll();
+    }
+
+    public CommentEntity addNewComment(CommentDto commentDto) {
+        CommentEntity comment = new CommentEntity();
+        comment.setContent(commentDto.getContent());
+        ProductEntity product = productRepository.getReferenceById(commentDto.getProductId());
+        comment.setProduct(product);
+        return commentRepository.save(comment);
+    }
+
+    public List<CommentEntity> deleteComment(Integer commentID) {
+        boolean exists = commentRepository.existsById(commentID);
+        if (!exists) {
+            throw new IllegalStateException("comment with id " + commentID + " does not exist");
```

Comment:
Ai putea crea exceptiile tale, pentru ca in loguri sa se vada dodata care serviciu a dat eroare. De exemplu CommentException, si faci fix aceasta dar acum throw new CommentException("message"). Sau chiar poti ceva mai concret, dar deseori nu e nevoie.

public class CommentException extends RuntimeException{ 
    public CommentException (String errorMessage) {
        super(errorMessage);
    }
}

Se recomanda de folosit RuntimeException in log de Exception, deoarece prima e uncatched (nu e nevoie de pus in try-catch)